src/core/services/theming/theming.js

```diff
@@ -1003,6 +1000,49 @@ function generateAllThemes($injector, $mdTheming) {
       }
     });
   }
+
+  // Split a string representing a CSS file producing an array with a rule in each element of it.
+  function splitCss(themeCss) {
+    var result = [];
+    var currentRule = '';
+    var openedCurlies = 0;
+    var closedCurlies = 0;
+
+    for (var i = 0; i < themeCss.length; i++) {
+      var character = themeCss.charAt(i);
+
+      // Check for content in quotes
+      if (character === '\'' || character === '\"') {
+        var pattern = '\\' + character + '.*' + '\\' + character;
+        var regex = new RegExp(pattern);
+        // Get content in quotes
+        var textInQuotes = regex.exec(themeCss.slice(i)).toString();
```

Comment:
Using `themeCss.indexOf(character, i+1)` to find the end quote would be faster than splitting the string and using a regex expression.